features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -332,6 +335,158 @@ class RumSessionEndedMetricTest {
         assertThat(viewCounts[SessionEndedMetric.VIEW_COUNTS_APP_LAUNCH_KEY]).isEqualTo(appLaunchViewCount)
     }
 
+    @Test
+    fun `M add count correctly W onUploadQualityTracked`(forge: Forge) {
+        // Given
+        val sessionEndedMetric = stubSessionEndedMetric()
+        val fakeTrack = forge.anElementFrom(
+            listOf(
+                "rum",
+                "logs",
+                "tracing",
+                "session-replay",
+                "session-replay-resources"
+            )
+        )
+        val fakeFirstEvent = UploadQualityEvent(
+            track = fakeTrack,
+            category = UploadQualityCategories.COUNT,
+            specificType = null
+        )
+
+        // When
+        sessionEndedMetric.onUploadQualityTracked(fakeFirstEvent)
+
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes(fakeNtpOffsetAtEnd)
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val uploadQuality = rse[UploadQualityMetric.UPLOAD_METRIC_KEY.key] as Map<*, *>
+        val eventTrack = uploadQuality[fakeTrack] as Map<*, *>
+        val cycleCount = eventTrack[UploadQualityMetric.CYCLE_COUNT_KEY.key] as Int
+        assertThat(cycleCount).isEqualTo(1)
+        val failures = eventTrack[UploadQualityMetric.FAILURE_COUNT_KEY.key] as Map<*, *>
+        assertThat(failures).isEqualTo(emptyMap<String, Any>())
```

Comment:
```suggestion
        assertThat(failures).isEmpty()
```

same can be applied to the similar lines below